tests/0099-commit_metadata.c

```diff
@@ -0,0 +1,193 @@
+                              rd_kafka_topic_partition_list_t *list,
+                              void *opaque) {
+        commit_cb_cnt++;
+        TEST_ASSERT(!err, "offset_commit_cb failure: %s",
+                    rd_kafka_err2str(err));
+}
+
+
+static void
+commit_metadata (const char *group_id,
+                 const rd_kafka_topic_partition_list_t *toppar_to_commit) {
+        rd_kafka_resp_err_t err;
+        rd_kafka_t *rk;
+        rd_kafka_conf_t *conf;
+
+        test_conf_init(&conf, NULL, 20/*timeout*/);
+
+        test_conf_set(conf, "group.id", group_id);
+
+        rd_kafka_conf_set_offset_commit_cb(conf, offset_commit_cb);
+
+        /* Create kafka instance */
+        rk = test_create_handle(RD_KAFKA_CONSUMER, conf);
+
+        TEST_SAY("Committing:\n");
+        print_toppar_list(toppar_to_commit);
+
+        err = rd_kafka_commit(rk, toppar_to_commit, 0);
+        TEST_ASSERT(!err, "rd_kafka_commit failed: %s", rd_kafka_err2str(err));
+
+        while (commit_cb_cnt == 0)
+                rd_kafka_poll(rk, 1000);
+
+        rd_kafka_destroy(rk);
+}
+
+
+static void
+get_committed_metadata (const char *group_id,
+                        const rd_kafka_topic_partition_list_t *toppar_to_check,
+                        const rd_kafka_topic_partition_list_t *expected_toppar) {
+        rd_kafka_resp_err_t err;
+        rd_kafka_t *rk;
+        rd_kafka_conf_t *conf;
+        rd_kafka_topic_partition_list_t *committed_toppar;
+
+        test_conf_init(&conf, NULL, 20/*timeout*/);
+
+        test_conf_set(conf, "group.id", group_id);
+
+        committed_toppar = rd_kafka_topic_partition_list_copy(toppar_to_check);
+
+        /* Create kafka instance */
+        rk = test_create_handle(RD_KAFKA_CONSUMER, conf);
+
+        err = rd_kafka_committed(rk, committed_toppar, tmout_multip(5000));
+        TEST_ASSERT(!err, "rd_kafka_committed failed: %s",
+                    rd_kafka_err2str(err));
+
+        compare_toppar_lists(committed_toppar, expected_toppar);
+
+        rd_kafka_destroy(rk);
+}
+
+int main_0099_commit_metadata (int argc, char **argv) {
+        rd_kafka_topic_partition_list_t *origin_toppar;
+        rd_kafka_topic_partition_list_t *expected_toppar;
+        const char *topic = test_mk_topic_name("0099-commit_metadata", 0);
+        const char *group_id = topic;
+
+        test_conf_init(NULL, NULL, 20/*timeout*/);
+
+        test_create_topic(topic, 1, 1);
+
+        origin_toppar = rd_kafka_topic_partition_list_new(1);
+
+        rd_kafka_topic_partition_list_add(origin_toppar, topic, 0);
+
+        expected_toppar = rd_kafka_topic_partition_list_copy(origin_toppar);
+
+        expected_toppar->elems[0].offset = 42;
+        expected_toppar->elems[0].metadata = rd_strdup("Hello world!");
+        expected_toppar->elems[0].metadata_size =
+                strlen(expected_toppar->elems[0].metadata);
+
+        get_committed_metadata(group_id, origin_toppar, origin_toppar);
```

Comment:
the logic here is convoluted (having the comparison assert inside the `get_committed_metadata` method, but it's just a test so whatever floats your boat)

Reply:
also nice test, but i don't understand why it's in this PR